CovertInfiltration/Src/CovertInfiltration/Classes/X2EventListener_Infiltration.uc

```diff
@@ -412,4 +442,70 @@ function XComReinforcementsDelayedVisualizationFn(XComGameState VisualizeGameSta
 
 	MessageBanner = X2Action_PlayMessageBanner(class'X2Action_PlayMessageBanner'.static.AddToVisualizationTree(ActionMetadata, VisualizeGameState.GetContext()));
 	MessageBanner.AddMessageBanner(default.strReinforcementDelayBannerMessage, , default.strReinforcementDelayBannerSubtitle, default.strReinforcementDelayBannerValue, eUIState_Good);
+}
+
+static function EventListenerReturn IncomingReinforcementsDisplay(Object EventData, Object EventSource, XComGameState GameState, Name EventID, Object CallbackData)
+{
+	local XComGameState_AIReinforcementSpawner ReinforcementSpawner;
+	local XComLWTuple Tuple;
+
+	Tuple = XComLWTuple(EventData);
+
+	foreach `XCOMHISTORY.IterateByClassType(class'XComGameState_AIReinforcementSpawner', ReinforcementSpawner)
+	{
+		break;
+	}
+
+	if (ReinforcementSpawner == None || Tuple == None)
+	{
+		return ELR_NoInterrupt;
+	}
+
+	Tuple.Data[0].b = true;
+	
+	if (ReinforcementSpawner.Countdown > 2)
+	{
+		Tuple.Data[1].s = class'UIUtilities_Text'.static.GetColoredText("REINFORCEMENTS", eUIState_Good);
+		Tuple.Data[2].s = class'UIUtilities_Text'.static.GetColoredText("INCOMING", eUIState_Good);
+		Tuple.Data[3].s = "828282";
+	}
+	else if (ReinforcementSpawner.Countdown == 2)
+	{
+		Tuple.Data[1].s = class'UIUtilities_Text'.static.GetColoredText("REINFORCEMENTS", eUIState_Warning);
+		Tuple.Data[2].s = class'UIUtilities_Text'.static.GetColoredText("WARNING", eUIState_Warning);
+		Tuple.Data[3].s = "fdce2b";
+	}
+	else
+	{
+		Tuple.Data[1].s = class'UIUtilities_Text'.static.GetColoredText("REINFORCEMENTS", eUIState_Bad);
+		Tuple.Data[2].s = class'UIUtilities_Text'.static.GetColoredText("IMMINENT", eUIState_Bad);
+		Tuple.Data[3].s = "bf1e2e";
+	}
+
+	return ELR_NoInterrupt;
+}
+
+static function EventListenerReturn DisableReinforcementsFlare(Object EventData, Object EventSource, XComGameState GameState, Name EventID, Object CallbackData)
+{
+	local XComGameState_AIReinforcementSpawner ReinforcementSpawner;
+	local XComLWTuple Tuple;
+
+	ReinforcementSpawner = XComGameState_AIReinforcementSpawner(EventSource);
+	Tuple = XComLWTuple(EventData);
+
+	if (ReinforcementSpawner == None || Tuple == None)
+	{
+		return ELR_NoInterrupt;
+	}
+
+	if (ReinforcementSpawner.Countdown > 1)
+	{
+		Tuple.Data[0].b = true;
+	}
+	else
+	{
+		Tuple.Data[0].b = false;
+	}
```

Comment:
This doesn't work the way you expect - it will prevent the flare from showing initially, and only if game is reloaded at `ReinforcementSpawner.Countdown == 1`, they will show

Reply:
well that's shitty is there any way to make it work how i want?